File: 10. Log/Log.py
import cv2
import numpy as np
from matplotlib import pyplot as plt

img = cv2.imread("log.tif", 0)
c = 255/np.log(1 + np.max(img))
log_pic = c * (np.log(1 + img))
log_pic = np.array(log_pic, dtype=np.uint8)
cv2.imshow("log", log_pic)

File: 10. Log/Log.m
img = imread('log.tif');
log_img = 0.1 * log(1 + double(img));
imshow(log_img);
log_img = 0.5 * log(1 + double(img));
imshow(log_img);
log_img = 0.05 * log(1 + double(img));
imshow(log_img);
figure;
imshow(img);
img = imread('gamma.tif')
new_img = imadjust(img, [], [], 3);
imshow(new_img);
figure;
imshow(img);
new_img = imadjust(img, [], [], 2);
imshow(new_img);